packages/components/src/components/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,172 @@
+import React from "react";
+import { SystemProp, Theme } from "@xstyled/styled-components";
+import { Box } from "../../primitives/Box";
+import { useGetImageOrientation } from "./useGetImageOrientation";
+import { Orientation } from "./getImageOrientation";
+import { getInitials } from "./getInitials";
+
+type AvatarSizeOptions = 6 | 8 | 10 | 12 | 14 | 16;
+
+type AvatarProps = {
+  /** Sets the image URL. */
+  src?: string;
+  /** Sets the alternative text. It is used to define initials if src isn't provided or fails to load and to add an accessible label. */
+  alt?: string;
+  /** Sets the avatar size. */
+  size?: AvatarSizeOptions;
+};
+
+const getAvatarSizes = (
+  size: AvatarSizeOptions
+): {
+  w: string;
+  h: string;
+  margin: SystemProp<keyof Theme["space"], Theme>;
+} => {
+  switch (size) {
+    case 6: {
+      return {
+        w: "24px",
+        h: "24px",
+        margin: "space30", // Should be 10px - no token provided
+      };
+    }
+    case 8: {
+      return {
+        w: "32px",
+        h: "32px",
+        margin: "space40",
+      };
+    }
+    case 10: {
+      return {
+        w: "40px",
+        h: "40px",
+        margin: "space50",
+      };
+    }
+    case 12: {
+      return {
+        w: "48px",
+        h: "48px",
+        margin: "space50",
+      };
+    }
+    case 14: {
+      return {
+        w: "56px",
+        h: "56px",
+        margin: "space50",
+      };
+    }
+    default: {
+      return {
+        w: "64px",
+        h: "64px",
+        margin: "space50",
+      };
+    }
+  }
+};
+
+const getInitialsSizes = (
+  size: AvatarSizeOptions
+): {
+  fontSize: SystemProp<keyof Theme["fontSizes"], Theme>;
+} => {
+  switch (size) {
+    case 6: {
+      return {
+        fontSize: "fontSize10",
+      };
+    }
+    case 8: {
+      return {
+        fontSize: "fontSize20",
+      };
+    }
+    case 10:
+    case 12: {
+      return {
+        fontSize: "fontSize30",
+      };
+    }
+    // Case 12: {
+    //   Return {
+    //     FontSize: "fontSize30", // Should be 18px - no token provided
```

Comment:
@richbachman same regarding the token here.

Reply:
Same here. Lets just question design on it.